src/averageAges.js

```diff
@@ -37,7 +42,33 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  let foundWoman = [];
+  let yearOfLife = 0;
+  const motherWithChildren = [];
+
+  foundWoman = people
+    .filter(woman => woman.sex === 'f');
+
+  foundWoman.forEach(element => {
+    if (people.some(woman => element.name === woman.mother)) {
+      motherWithChildren.push(element);
+    }
+  });
+
+  yearOfLife = withChildren
+    ? motherWithChildren
+      .map(year => year.died - year.born)
+      .reduce((a, b) => a + b)
+
+    : foundWoman
+      .map(year => year.died - year.born)
+      .reduce((a, b) => a + b);
```

Comment:
Don't repeat yourself: just set correct array to the one variable and use it in your code

Reply:
You didn't fix my comment 🥲